ext/standard/array.c

```diff
@@ -345,7 +345,24 @@ static zend_always_inline int php_array_key_compare_string_locale_unstable_i(Buc
 
 static zend_always_inline int php_array_data_compare_unstable_i(Bucket *f, Bucket *s) /* {{{ */
 {
-	return zend_compare(&f->val, &s->val);
+	int result = zend_compare(&f->val, &s->val);
+	// Special handling for enums
```

Comment:
Before merging, add a comment that this is added for array_unique. Otherwise this may seem cryptic.